sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_client.py

```diff
@@ -0,0 +1,944 @@
+
+    def backup_certificate(self, name, **kwargs):
+        # type: (str) -> bytes
+        """Backs up the specified certificate.
+
+        Requests that a backup of the specified certificate be downloaded
+        to the client. All versions of the certificate will be downloaded.
+        This operation requires the certificates/backup permission.
+
+        :param name: The name of the certificate.
+        :type name: str
+        :return: the backup blob containing the backed up certificate.
+        :rtype: bytes
+        :raises:
+         :class:`KeyVaultErrorException<azure.keyvault.v7_0.models.KeyVaultErrorException>`
+        """
+        backup_result = self._client.backup_certificate(
+            vault_base_url=self.vault_url,
+            certificate_name=name,
+            error_map={404: ResourceNotFoundError},
+            **kwargs
+        )
+        return backup_result.value
+
+
+    def restore_certificate(self, backup, **kwargs):
+        # type: (bytes) -> Certificate
+        """Restores a backed up certificate to a vault.
+
+        Restores a backed up certificate, and all its versions, to a vault.
+        this operation requires the certificates/restore permission.
+
+        :param backup: The backup blob associated with a certificate bundle.
+        :type backup bytes
+        :return: The restored Certificate
+        :rtype: ~azure.security.keyvault.certificates._models.Certificate
+        :raises:
+         :class:`KeyVaultErrorException<azure.keyvault.v7_0.models.KeyVaultErrorException>`
+        """
+        bundle = self._client.restore_certificate(
+            vault_base_url=self.vault_url,
+            certificate_bundle_backup=backup,
+            error_map={409: ResourceExistsError},
+            **kwargs
+        )
+        return Certificate._from_certificate_bundle(certificate_bundle=bundle)
+
+    def list_deleted_certificates(self, include_pending=None, **kwargs):
+        # type: (Optional[bool]) -> Generator[DeletedCertificate]
+        """Lists the deleted certificates in the specified vault currently
+        available for recovery.
+
+        Retrieves the certificates in the current vault which are in a deleted
+        state and ready for recovery or purging. This operation includes
+        deletion-specific information. This operation requires the certificates/get/list
+        permission. This operation can only be enabled on soft-delete enabled vaults.
+
+        :param include_pending: Specifies whether to include certificates which are not
+        completely provisioned.
+        :type include_pending bool
+        :return: An iterator like instance of DeletedCertificate
+        :rtype:
+         typing.Generator[~azure.security.keyvault.certificates._models.DeletedCertificate]
+        :raises:
+         :class:`KeyVaultErrorException<azure.keyvault.v7_0.models.KeyVaultErrorException>`
+        """
+        max_page_size = kwargs.get("max_page_size", None)
```

Comment:
This should be `kwargs.pop`. `get` doesn't remove the item from the dict, which is then passed on to other code which is likely to fail due to the unexpected kwarg.